lib/blockchain.go

```diff
@@ -2115,24 +2190,26 @@ func (bc *Blockchain) processHeaderPoW(blockHeader *MsgDeSoHeader, headerHash *B
 			"before timestamp of previous block %v",
 			time.Unix(int64(blockHeader.GetTstampSecs()), 0),
 			time.Unix(int64(parentHeader.GetTstampSecs()), 0))
-		return false, false, HeaderErrorTimestampTooEarly
+		return nil, false, false, HeaderErrorTimestampTooEarly
 	}
 
+	// @diamondhands - this check is significantly slower because we need to go
```

Comment:
@diamondhands0 - header processing for PoW blocks is significantly slower. Just a heads up. nice thing is that this is a fixed cost on each network. It took about 7m to sync all PoW headers (1.2 M headers) on testnet, so it's not *terrible*.

Reply:
I think that's fine. We could do something to skip this check, but I'm worried it would expose us to attack so better to just leave it for now, and optimize it later if it just hits us when syncing a node from scratch.

Reply:
Another way to optimize it would be to keep a map from height to "node in the best chain at that height" in the blockindex. But no need to add that complexity now. It would need to be pruned and stuff as the LRU cache fills. Gross.

Reply:
yeah exactly. it's gross 😂 we could add a new prefix to the DB to do it, but it does not make sense to do it in memory b/c eventually we'll need to move it to badger.